x/auth/client/cli/multisign.go

```diff
@@ -0,0 +1,161 @@
+multisig key <name>, and attach it to the transaction read from <file>.
+
+If the flag --signature-only flag is on, it outputs a JSON representation
+of the generated signature only.
+
+The --offline flag makes sure that the client will not reach out to an external node.
+Thus account number or sequence number lookups will not be performed and it is
+recommended to set such parameters manually.`,
+		RunE: makeMultiSignCmd(codec),
+		Args: cobra.MinimumNArgs(3),
+	}
+	cmd.Flags().Bool(flagSigOnly, false, "Print only the generated signature, then exit")
+	cmd.Flags().Bool(flagOffline, false, "Offline mode. Do not query a full node")
+	cmd.Flags().String(flagOutfile, "",
+		"The document will be written to the given file instead of STDOUT")
+
+	// Add the flags here and return the command
+	return client.PostCommands(cmd)[0]
+}
+
+func makeMultiSignCmd(cdc *amino.Codec) func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) (err error) {
+		stdTx, err := readAndUnmarshalStdTx(cdc, args[0])
+		if err != nil {
+			return
+		}
+
+		keybase, err := keys.GetKeyBaseFromDir(viper.GetString(cli.HomeFlag))
+		if err != nil {
+			return
+		}
+
+		multisigInfo, err := keybase.Get(args[1])
+		if err != nil {
+			return
+		}
+		if multisigInfo.GetType() != crkeys.TypeOffline {
+			return fmt.Errorf("%q must be of type offline: %s",
+				args[1], multisigInfo.GetType())
+		}
+
+		multisigPub := multisigInfo.GetPubKey().(multisig.PubKeyMultisigThreshold)
+		multisigSig := multisig.NewMultisig(len(multisigPub.PubKeys))
+		cliCtx := context.NewCLIContext().WithCodec(cdc).WithAccountDecoder(cdc)
+		txBldr := authtxb.NewTxBuilderFromCLI()
+
+		if !viper.GetBool(flagOffline) {
+			addr := multisigInfo.GetAddress()
+			accnum, err := cliCtx.GetAccountNumber(addr)
+			if err != nil {
+				return err
+			}
+
+			seq, err := cliCtx.GetAccountSequence(addr)
+			if err != nil {
+				return err
+			}
+
+			txBldr = txBldr.WithAccountNumber(accnum).WithSequence(seq)
+		}
+
+		// read each signature and add it to the multisig if valid
+		for i := 2; i < len(args); i++ {
+			stdSig, err := readAndUnmarshalStdSignature(cdc, args[i])
+			if err != nil {
+				return err
+			}
+
+			// Validate each signature
+			sigBytes := auth.StdSignBytes(
+				txBldr.GetChainID(), txBldr.GetAccountNumber(), txBldr.GetSequence(),
+				stdTx.Fee, stdTx.GetMsgs(), stdTx.GetMemo(),
+			)
+			if ok := stdSig.PubKey.VerifyBytes(sigBytes, stdSig.Signature); !ok {
+				return fmt.Errorf("couldn't verify signature")
+			}
+			multisigSig.AddSignatureFromPubKey(stdSig.Signature, stdSig.PubKey, multisigPub.PubKeys)
+		}
+
+		newStdSig := auth.StdSignature{Signature: cdc.MustMarshalBinaryBare(multisigSig), PubKey: multisigPub}
+		newTx := auth.NewStdTx(stdTx.GetMsgs(), stdTx.Fee, []auth.StdSignature{newStdSig}, stdTx.GetMemo())
+
+		var json []byte
+		switch viper.GetBool(flagSigOnly) {
```

Comment:
Since `viper.GetBool(flagSigOnly)` and `cliCtx.Indent` are both booleans, I think simple if/else blocks will be cleaner and less code 👍

Reply:
How about:

```golang
		var json []byte
		switch viper.GetBool(flagSigOnly) {
		case true && cliCtx.Indent:
			json, err = cdc.MarshalJSONIndent(newTx.Signatures[0], "", "  ")
		case true && !cliCtx.Indent:
			json, err = cdc.MarshalJSON(newTx.Signatures[0])
		case false && cliCtx.Indent:
			json, err = cdc.MarshalJSONIndent(newTx, "", "  ")
		default:
			json, err = cdc.MarshalJSON(newTx)
		}
		if err != nil {
			return err
		}
```

Reply:
haha sure!